infrastructure/data/index.ts

```diff
@@ -12,8 +12,9 @@ const networking = new pulumi.StackReference(`planx/networking/${env}`);
 
 const db = new aws.rds.Instance("app", {
   engine: "postgres",
-  // Available versions: https://docs.aws.amazon.com/AmazonRDS/latest/UserGuide/CHAP_PostgreSQL.html#PostgreSQL.Concepts.General.version12
-  engineVersion: "12.3",
+  // Available versions:
+  // $ aws rds describe-db-engine-versions --default-only --engine postgres
+  engineVersion: "14.6",
```

Comment:
Can't use `12.3` anymore as it's been deprecated. This diff isn't going to be a problem for staging nor production as we usually only deploy the `application` stack.